payroll/models/hr_payslip.py

```diff
@@ -711,55 +747,73 @@ def get_payslip_vals(
         )
         return res
 
-    @api.onchange("employee_id", "date_from", "date_to", "struct_id")
-    def onchange_employee(self):
+    def _get_employee_contracts(self):
+        return self.env["hr.contract"].browse(
+            self.employee_id._get_contracts(
+                date_from=self.date_from, date_to=self.date_to
+            ).ids
+        )
 
-        if (not self.employee_id) or (not self.date_from) or (not self.date_to):
+    @api.onchange("struct_id")
+    def onchange_struct_id(self):
+        if not self.struct_id:
             return
```

Comment:
It doesn't make sense to me to just return here and leave the previous payslip lines and inputs. If struct_id is false all the existing payslip lines and inputs should be removed.

Reply:
That's the way it was implemented in the module the first time. 

I only added that change because if you don't return the struct_id will be replaced by the contract struct_id every time you change the date, or the employee.

From a user perspective you will select a struct_id and then change the date for example, in those cases you don't want the struct_id to be changed by the default one. That's why I added that return. 

Then if everyone thinks it's a good idea to remove the computer lines in those cases I will do it. But the module itself from the beginning never remove any line if you don't click the compute button. I think it's okay to prevent deleting manual changes. If the user want to recompute, can press the button again. 

But I leave this opened for more opinions.

Reply:
Also if struct_id is false, then yes it will be fetched from the contract default one. That return is for the cases where it's manually selected. If it's not selected it will be defaulted.

Reply:
Understood. Thanks.